### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by jschwartz73** [jhuckaby/Cronicle#22]
I'm a big fan of Cronicle and would like to use it on my current project.  However, the customer is requiring that deployment be done on Windows.

How difficult do you think it would be for a novice Cronicle user to add windows support (which of course would be contributed back)?

**Comment by jhuckaby** [comment]
Hi there!  Thanks for being a fan 😃 

Unfortunately, I have no idea what it would take to add Windows support in Cronicle.  I suspect there would be lots of changes required, including but not limited to, the way it spawns processes to run jobs, especially detached jobs, and the way processes talk back and forth to each other using Unix STDOUT pipes.  I don't know if Windows even has such a thing.  Also, resource monitoring uses a very Unix `ps -eo` command, and I am fairly sure there are a lot of hard-coded `/` slashes for file paths in various places.

Suffice to say it would probably be a rather large undertaking, but you are welcome to give it a try.  That's what's so great about MIT open source software on GitHub!  Fork and go!

**Comment by jschwartz73** [comment]
Thanks for the update.  I'm not going to be able to put the time in to getting Cronicle to work on Windows via the source code, but for those interested, the following information will let you know what I tried:

Ubuntu Bash Shell for Windows - This windows 10 feature gives a bash shell to enable development on Windows.  They label this as a developer feature.  I got the Bash shell installed on Win 10 Anniversary edition, but kept having problems getting cronicle to start because of a call to get the network interfaces.  I tracked that down and found the Windows 10 Creator Edition resolved the problem.  From there i was able to run control.sh setup and it worked.  Then i was able to call control.sh start which also worked.  The problem i had was there was no way to route traffic to the Bash shell.  The IP address of the bash shell was the same as the host, there was no firewall, but still could not find a way to route traffic to Cronicle.  After more research, I found this statement on Microsoft's website (**emphasis mine**)

> This is a developer toolset to help you write and build all your code for all your scenarios and platforms.” It’s not a full Ubuntu virtual machine. **You can’t use it to host servers, as you could on Linux**. If you need that, you can always install Linux in a virtual machine using Microsoft’s Hyper-V technology, VirtualBox, VMware, or something similar.

Cygwin - I got cygwin installed, but node never worked properly in it.
